api/src/services/observation-service.ts

```diff
@@ -50,4 +77,97 @@ export class ObservationService extends DBService {
   async getSurveyObservations(surveyId: number): Promise<ObservationRecord[]> {
     return this.observationRepository.getSurveyObservations(surveyId);
   }
+
+  /**
+   * Inserts a survey observation submission record into the database and returns the key
+   *
+   * @param {Express.Multer.File} file
+   * @param {number} projectId
+   * @param {number} surveyId
+   * @return {*}  {Promise<{ key: string }>}
+   * @memberof ObservationService
+   */
+  async insertSurveyObservationSubmission(
+    file: Express.Multer.File,
+    projectId: number,
+    surveyId: number
+  ): Promise<{ submission_id: number; key: string }> {
+    const submissionId = await this.observationRepository.getNextSubmissionId();
+
+    const key = generateS3FileKey({
+      projectId: projectId,
+      surveyId: surveyId,
+      submissionId: submissionId,
+      fileName: file.originalname
+    });
+
+    const insertResult = await this.observationRepository.insertSurveyObservationSubmission(
+      submissionId,
+      key,
+      surveyId,
+      file.originalname
+    );
+
+    return { submission_id: insertResult.submission_id, key };
```

Comment:
This may be correct in shorthand for the `key` param just wanted to double check the return type is correct

Reply:
This is correct